python/CMakeLists.txt

```diff
@@ -16,34 +19,48 @@ else()
   target_link_libraries(pyf3d PRIVATE libf3d)
 endif()
 
+set(f3d_python_package_name "f3d")
+set(f3d_python_package_suffix "python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages")
+set(f3d_python_lib_output "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${f3d_python_package_suffix}/${f3d_python_package_name}")
+
 set_target_properties(pyf3d PROPERTIES
   CXX_STANDARD 14
   CXX_VISIBILITY_PRESET hidden
-  LIBRARY_OUTPUT_NAME "f3d"
+  LIBRARY_OUTPUT_DIRECTORY "${f3d_python_lib_output}"
   )
 
-if(WIN32)
-  # On Windows, the python module needs to be in the same folder than f3d.dll
-  # Usage of PATH to find the DLL is not possible, see https://stackoverflow.com/a/64303856/2609654
-  set_target_properties(pyf3d PROPERTIES
-    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}"
-    )
+if (APPLE)
+  set_target_properties(pyf3d PROPERTIES INSTALL_RPATH "@loader_path/../../../${CMAKE_INSTALL_LIBDIR}")
+elseif (UNIX)
+  set_target_properties(pyf3d PROPERTIES INSTALL_RPATH "$ORIGIN/../../../${CMAKE_INSTALL_LIBDIR}")
 endif()
+set(f3d_python_install_path "${CMAKE_INSTALL_LIBDIR}/${f3d_python_package_suffix}/${f3d_python_package_name}")
+set(f3d_python_dll_install_bin_path "../../../${CMAKE_INSTALL_BINDIR}")
+
+# generate fixup for build
+f3d_python_windows_dll_fixup(ABSOLUTE_PATHS "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}" "$<TARGET_FILE_DIR:VTK::CommonCore>" OUTPUT F3D_PYTHON_FIXUP)
+
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/__init__.py.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/__init__.py-build" @ONLY)
 
-if (APPLE OR UNIX)
-  set_target_properties(pyf3d PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
-endif ()
+file(GENERATE
+  OUTPUT "${f3d_python_lib_output}/__init__.py"
+  INPUT "${CMAKE_CURRENT_BINARY_DIR}/__init__.py-build"
+)
+
+# generate fixup for install
+f3d_python_windows_dll_fixup(RELATIVE_PATHS "${f3d_python_dll_install_bin_path}" OUTPUT F3D_PYTHON_FIXUP)
```

Comment:
would it be better to check  for win32 outside of the method ? that makes it clearer imo.